internal/cri/server/podsandbox/controller.go

```diff
@@ -130,7 +132,8 @@ type Controller struct {
 	// actually we only use it's backoff mechanism to make sure pause container is cleaned up.
 	eventMonitor *events.EventMonitor
 
-	store *Store
+	store  *Store
+	locker kmutex.KeyedLocker
```

Comment:
pls add a comment regarding the scope and purpose of the controller's sb.ID keyd locks vs the store's internal m sync.Map 

it feels awkward to have a key'd mutex between request to stop operations and task exit originated removal operations

if feels like we are trying to protect the use of store keys (pod ids) that we got from the store while they are in the process of being removed...

hmm...

maybe if the key'd mutex was in the store and it protected a field called sandboxBeing removed... you would have a GetForRemoval and that would tell you if you are first trying to remove this sandbox id or not...